add merge function for NamedDataStore

lucylq opened the pull request:
Summary:
Allow us to change the NamedDataStore model. Usage will now be:
```
named_data_store
for backend in backends:
    bytes, named_data_store = backend.preprocess()
    named_data_store.merge(named_data_store)
```

Note:
- named_data_store is held by edge program manager
- preprocess returns the named data store as part of PreprocessResult

Differential Revision: D70409078

mcr229 commented on exir/_serialize/test/test_named_data_store.py:
```diff
@@ -83,3 +83,41 @@ def test_add_duplicate_key_fail(self) -> None:
         self.assertEqual(output.pte_data["key"], 0)
         self.assertEqual(len(output.external_data), 0)
+
+    def test_merge_named_data_store(self) -> None:
```
can we just add one negative test case where two NamedDataStore have the same key but different data, and expect the error thrown?

lucylq replied:
thanks, updated!

lucylq committed:
add merge function for NamedDataStore (#8850)

Summary:

Allow us to change the NamedDataStore model. Usage will now be:
```
named_data_store
for backend in backends:
    bytes, named_data_store = backend.preprocess()
    named_data_store.merge(named_data_store)
```

Note:
- named_data_store is held by edge program manager
- preprocess returns the named data store as part of PreprocessResult

Differential Revision: D70409078
